.github/workflows/build-revive-wasm.yml

```diff
@@ -0,0 +1,83 @@
+name: Build revive-wasm
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  REVIVE_WASM_INSTALL_DIR: ${{ github.workspace }}/target/wasm32-unknown-emscripten/release
+  EMSCRIPTEN_VERSION: 3.1.64
+
+jobs:
+  build-revive-wasm:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache LLVM build
+        id: cache-llvm
+        uses: actions/cache@v3
+        with:
+          path: |
+            llvm18.0-emscripten
+          # Use a unique key based on LLVM version or configuration files to avoid cache invalidation
+          key: llvm-build-${{ runner.os }}-${{ hashFiles('clone-llvm.sh', 'emscripten-build-llvm.sh') }}
+
+      - name: Install LLVM Dependencies
+        if: steps.cache-llvm.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get update && sudo apt-get install -y cmake ninja-build libncurses5 libtinfo-dev libtinfo5
+          # Install Emscripten
+          git clone https://github.com/emscripten-core/emsdk.git
+          cd emsdk
+          ./emsdk install ${{ env.EMSCRIPTEN_VERSION }}
+          ./emsdk activate ${{ env.EMSCRIPTEN_VERSION }}
+
+      - name: Install Dependencies
+        run: |
+          rustup target add wasm32-unknown-emscripten
+          # Install LLVM required for the compiler runtime, runtime-api and stdlib
+          curl -sSL --output llvm.tar.xz https://github.com/llvm/llvm-project/releases/download/llvmorg-18.1.4/clang+llvm-18.1.4-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+          tar Jxf llvm.tar.xz
+          mv clang+llvm-18.1.4-x86_64-linux-gnu-ubuntu-18.04 llvm18/
+          echo "$(pwd)/llvm18/bin" >> $GITHUB_PATH
```

Comment:
Sorry I still don't understand what we need it for. Don't we build LLVM completely from source with emscripten? What is this binary version needed for? Shouldn't the emscripten build version be used for everything?

Reply:
We need the RISC-V compiler-rt and also we compile some shared code (for example our "stdlib") into LLVM bitcode at this stage. Those bitcodes are bundled with the compiler and just linked into into contract code. The emscripten version does compile contract code but not those things.

Reply:
These crates are compiled on the host to the RISC-V target, to be linked into the contract binary when it is built using the WASM version of revive.

Reply:
Thanks